File: lib/src/main/kotlin/dbassistant/layers/QueryPreparationLayer.kt
package dbassistant.layers

import dbassistant.interfaces.ILayer

class QueryPreparationLayer : ILayer  {

    override fun pass(cargo: LayerCargo): LayerCargo {
        val input = cargo.take() as List<String>
        val output = input.joinToString(" ")
        cargo.put(output)
        return cargo
    }

}

File: dbassistant/layers/QueryPreparationLayer.ts
import ILayer from '../interfaces/ILayer';
import LayerCargo from './LayerCargo';

class QueryPreparationLayer implements ILayer {
    process(cargo: LayerCargo): LayerCargo {
        const input = cargo.take();
        if (typeof input[0] !== 'string') {
            throw new Error('Input should be a string');
        }
        const output = input.join(' ');
        cargo.put(output);
        return cargo;
    }
}

export default QueryPreparationLayer;